crates/provider-sdk/src/provider.rs

```diff
@@ -94,18 +103,23 @@ impl ProviderConnection {
     pub(crate) fn new(
         nats: async_nats::Client,
         host_data: &HostData,
+        wit_resolve: Option<Resolve>,
     ) -> ProviderInitResult<ProviderConnection> {
         let key = Arc::new(
             KeyPair::from_seed(&host_data.invocation_seed)
                 .map_err(|e| ProviderInitError::Initialization(format!("key failure: {e}")))?,
         );
 
+        // Shared reference to the WIT that powers this provider
+        let wit_resolve = Arc::new(wit_resolve);
```

Comment:
Just a note that this should not exist at runtime (in the future?). The macro will have to derive `wrpc::Type` values using resolve at compile-time, but only the function parameter and result tuple types should persist

Reply:
OK awesome, once `wrpc` exists, we can change this code here!